api/service/explorer/service.go

```diff
@@ -258,6 +266,52 @@ func (s *Service) GetExplorerTransaction(w http.ResponseWriter, r *http.Request)
 	data.TX = *tx
 }
 
+// GetCommittee servers /comittee end-point.
+func (s *Service) GetCommittee(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	shardIDRead := r.FormValue("shard_id")
+	epochRead := r.FormValue("epoch")
+	shardID := uint64(0)
+	epoch := uint64(0)
+	var err error
+	if shardIDRead != "" {
+		shardID, err = strconv.ParseUint(shardIDRead, 10, 32)
+		if err != nil {
+			utils.Logger().Warn().Err(err).Msg("cannot read shard id")
+		}
+	}
+	if epochRead != "" {
+		epoch, err = strconv.ParseUint(epochRead, 10, 64)
+		if err != nil {
+			utils.Logger().Warn().Err(err).Msg("cannot read shard epoch")
+		}
+	}
+	if s.ShardID != uint32(shardID) {
+		utils.Logger().Warn().Err(err).Msg("incorrect shard id")
+		return
+	}
+	db := s.storage.GetDB()
+	bytes, err := db.Get([]byte(GetCommitteeKey(uint32(shardID), epoch)))
+	if err != nil {
+		utils.Logger().Warn().Err(err).Msg("cannot read committee")
+		return
+	}
+	committee := &types.Committee{}
+	if rlp.DecodeBytes(bytes, committee) != nil {
+		utils.Logger().Warn().Msg("cannot convert data from DB")
+		return
+	}
+	validators := &Committee{}
+	for _, validator := range committee.NodeList {
+		validatorStake := big.NewInt(0)
+		validatorStake, err = s.GetAccountBalance(validator.EcdsaAddress)
+		validators.Validators = append(validators.Validators, &Validator{Address: common2.MustAddressToBech32(validator.EcdsaAddress), Stake: validatorStake})
+	}
+	if err := json.NewEncoder(w).Encode(validators); err != nil {
+		utils.Logger().Warn().Err(err).Msg("cannot JSON-encode committee")
```

Comment:
When it happens, the JSON response written to `w` may be invalid or truncated.  Return a 500 error.